MekHQ/src/mekhq/campaign/personnel/divorce/AbstractDivorce.java

```diff
@@ -0,0 +1,270 @@
+    private boolean useRandomSameSexDivorce;
+    private boolean useRandomClannerDivorce;
+    private boolean useRandomPrisonerDivorce;
+
+    private final ResourceBundle resources = ResourceBundle.getBundle("mekhq.resources.Personnel", new EncodeControl());
+    //endregion Variable Declarations
+
+    //region Constructors
+    protected AbstractDivorce(final RandomDivorceMethod method, final CampaignOptions options) {
+        this.method = method;
+        setUseClannerDivorce(options.isUseClannerDivorce());
+        setUsePrisonerDivorce(options.isUsePrisonerDivorce());
+        setUseRandomOppositeSexDivorce(options.isUseRandomOppositeSexDivorce());
+        setUseRandomSameSexDivorce(options.isUseRandomSameSexDivorce());
+        setUseRandomClannerDivorce(options.isUseRandomClannerDivorce());
+        setUseRandomPrisonerDivorce(options.isUseRandomPrisonerDivorce());
+    }
+    //endregion Constructors
+
+    //region Getters/Setters
+    public RandomDivorceMethod getMethod() {
+        return method;
+    }
+
+    public boolean isUseClannerDivorce() {
+        return useClannerDivorce;
+    }
+
+    public void setUseClannerDivorce(final boolean useClannerDivorce) {
+        this.useClannerDivorce = useClannerDivorce;
+    }
+
+    public boolean isUsePrisonerDivorce() {
+        return usePrisonerDivorce;
+    }
+
+    public void setUsePrisonerDivorce(final boolean usePrisonerDivorce) {
+        this.usePrisonerDivorce = usePrisonerDivorce;
+    }
+
+    public boolean isUseRandomOppositeSexDivorce() {
+        return useRandomOppositeSexDivorce;
+    }
+
+    public void setUseRandomOppositeSexDivorce(final boolean useRandomOppositeSexDivorce) {
+        this.useRandomOppositeSexDivorce = useRandomOppositeSexDivorce;
+    }
+
+    public boolean isUseRandomSameSexDivorce() {
+        return useRandomSameSexDivorce;
+    }
+
+    public void setUseRandomSameSexDivorce(final boolean useRandomSameSexDivorce) {
+        this.useRandomSameSexDivorce = useRandomSameSexDivorce;
+    }
+
+    public boolean isUseRandomClannerDivorce() {
+        return useRandomClannerDivorce;
+    }
+
+    public void setUseRandomClannerDivorce(final boolean useRandomClannerDivorce) {
+        this.useRandomClannerDivorce = useRandomClannerDivorce;
+    }
+
+    public boolean isUseRandomPrisonerDivorce() {
+        return useRandomPrisonerDivorce;
+    }
+
+    public void setUseRandomPrisonerDivorce(final boolean useRandomPrisonerDivorce) {
+        this.useRandomPrisonerDivorce = useRandomPrisonerDivorce;
+    }
+    //endregion Getters/Setters
+
+    /**
+     * This is used to determine if a person can divorce.
+     * @param person the person to determine for
+     * @param randomDivorce if this is for random divorce or manual divorce
+     * @return null if they can, otherwise the reason they cannot
+     */
+    public @Nullable String canDivorce(final Person person, final boolean randomDivorce) {
```

Comment:
Same as with Marriage I do think this would benefit from being a boolean and an additional method delivering a cause.